### PR TITLE
[core] Added SRTO_MININPUTBW

### PR DESCRIPTION
**Pull request by maxsharabayko**
A simplification of #1541 and #1383.
The proposed `SRTO_MININPUTBW` socket option allows specifying the minimum input bitrate estimate used to set bandwidth limit.

Effective when `SRTO_INPUTBW = 0` -> input bitrate estimation.
```
input_rate = max(SRTO_MININPUTBW, estimate)
BWLimit = input_rate * (1 + SRTO_OHEADBW / 100.0)
```

### TODO

- [x] Update APISocketOptions.md

### CODE REVIEW

**Review by jeandube — COMMENTED** [review]
Way simpler than the SRTO_OUTPACEMODE option it replaces. Maintain backward compatibility but setting both SRTO_INPUTBW and SRTO_MININPUTBW would lead to unexpected results. The documentation should clarify that SRTO_MININPUTBW is a floor to the internally measured input rate.

File: srtcore/core.cpp [hunk 1]
```diff
@@ -545,6 +547,14 @@ void CUDT::setOpt(SRT_SOCKOPT optName, const void* optval, int optlen)
             updateCC(TEV_INIT, EventVariant(TEV_INIT_INPUTBW));
         break;
 
+    case SRTO_MININPUTBW:
+        m_llMinInputBW = cast_optval<int64_t>(optval, optlen);
+        // (only if connected; if not, then the value
+        // from m_iOverheadBW will be used initially)
```
[the remark this answers is not in the document]
Comment (maxsharabayko) [reply srtcore/core.cpp:553]: It is 🙂

**Comment by maxsharabayko** [comment]
> Way simpler than the SRTO_OUTPACEMODE option it replaces. Maintain backward compatibility but setting both SRTO_INPUTBW and SRTO_MININPUTBW would lead to unexpected results. The documentation should clarify that SRTO_MININPUTBW is a floor to the internally measured input rate.

The proposed behavior is that `SRTO_MININPUTBW` is only effective when both `SRTO_MAXBW` and `SRTO_INPUTBW` are set to zero, meaning input bitrate estimation is in use.

**Comment by maxsharabayko** [comment]
### Test

Sending starts at 1 Mbps, five seconds later sending rate is increased to 5 Mbps.

#### Input BW Estimation

```c++
SRTO_MAXBW = 0
SRTO_INPUTBW = 0; // Estimate internally
SRTO_OHEADBW = 25;
SRTO_MININPUTBW = 0;
```

sender-estinputbw

#### Input BW Estimation + Setting Min Input BW

```c++
SRTO_MAXBW = 0
SRTO_INPUTBW = 0; // Estimate internally
SRTO_OHEADBW = 25;
SRTO_MININPUTBW = 1 Mbps;
// Once "encoder" changes sending rate from 1Mbps to 5Mbps
SRTO_MININPUTBW = 5 Mbps;
```

setting-mininputbw

File: test/test_socket_options.cpp [hunk 2]
```diff
@@ -121,3 +135,114 @@ TEST_F(TestSocketOptions, LossMaxTTL)
 }
 
 
+// Try to set/get SRTO_MININPUTBW with wrong optlen
+TEST_F(TestSocketOptions, MinInputBWWrongLen)
+{
```
Comment (ethouris) [inline test/test_socket_options.cpp:140]: This test is kinda pointless. The size check is something we need actually for all options to be done as an overall solution.

Comment (maxsharabayko) [reply test/test_socket_options.cpp:140]: > The size check is something we need actually for all options to be done as an overall solution.
Absolutely! Issue #1819 is created to track that.

This PR adds checks for BW-related option, as they cast to `int64_t`. The chances of mistakenly passing an `int` are higher than e.g. passing an `unit16_t` to an option expecting `int`. But all other checks are better to be added as well.

**Review by ethouris — APPROVED** [review]

**Pull request merged**